metrics/prometheus/metrics.go

```diff
@@ -0,0 +1,68 @@
+package prometheus
+
+import (
+	"errors"
+	"time"
+
+	"github.com/micro/go-micro/v3/metrics"
+)
+
+// ErrPrometheusIsCruft is a terrible thing to have to do. The Prometheus client should really appreciate errors more:
+var ErrPrometheusIsCruft = errors.New("The venerable Prometheus client panicked. Did you do something like change the tag cardinality or the type of a metric?")
```

Comment:
Best. Error. Name. Ever.

Reply:
I need to check that this is still a thing, expect this to change to something less crufty.